pkg/apis/garden/v1beta1/conversions.go

```diff
@@ -1272,3 +1268,1571 @@ func volumeTypesHaveName(volumeTypes []garden.VolumeType, name string) bool {
 	}
 	return false
 }
+
+func Convert_v1beta1_Shoot_To_garden_Shoot(in *Shoot, out *garden.Shoot, s conversion.Scope) error {
+	if err := autoConvert_v1beta1_Shoot_To_garden_Shoot(in, out, s); err != nil {
+		return err
+	}
+
+	var networking garden.Networking
+	if in.Spec.Networking != nil {
+		if err := Convert_v1beta1_Networking_To_garden_Networking(in.Spec.Networking, &networking, s); err != nil {
+			return err
+		}
+	}
+	out.Spec.Networking = networking
+
+	var dns garden.DNS
+	if err := autoConvert_v1beta1_DNS_To_garden_DNS(&in.Spec.DNS, &dns, s); err != nil {
+		return err
+	}
+
+	var provider garden.DNSProvider
+	if in.Spec.DNS.ExcludeDomains != nil || in.Spec.DNS.ExcludeZones != nil || in.Spec.DNS.IncludeDomains != nil || in.Spec.DNS.IncludeZones != nil || in.Spec.DNS.Provider != nil || in.Spec.DNS.SecretName != nil {
+		provider.SecretName = in.Spec.DNS.SecretName
+		provider.Type = in.Spec.DNS.Provider
+
+		var domains *garden.DNSIncludeExclude
+		if in.Spec.DNS.IncludeDomains != nil || in.Spec.DNS.ExcludeDomains != nil {
+			domains = &garden.DNSIncludeExclude{}
+			for _, val := range in.Spec.DNS.IncludeDomains {
+				domains.Include = append(domains.Include, val)
+			}
+			for _, val := range in.Spec.DNS.ExcludeDomains {
+				domains.Exclude = append(domains.Exclude, val)
+			}
+		}
+		provider.Domains = domains
+
+		var zones *garden.DNSIncludeExclude
+		if in.Spec.DNS.IncludeZones != nil || in.Spec.DNS.ExcludeZones != nil {
+			zones = &garden.DNSIncludeExclude{}
+			for _, val := range in.Spec.DNS.IncludeZones {
+				zones.Include = append(zones.Include, val)
+			}
+			for _, val := range in.Spec.DNS.ExcludeZones {
+				zones.Exclude = append(zones.Exclude, val)
+			}
+		}
+		provider.Zones = zones
+
+		dns.Providers = append(dns.Providers, provider)
+	}
+
+	if additionalProviders, ok := in.Annotations[garden.MigrationShootDNSProviders]; ok {
+		var providers []garden.DNSProvider
+		if err := json.Unmarshal([]byte(additionalProviders), &providers); err != nil {
+			return err
+		}
+		dns.Providers = append(dns.Providers, providers...)
+	}
+	out.Spec.DNS = &dns
+
+	var workerMigrationInfo garden.WorkerMigrationInfo
+	if data, ok := in.Annotations[garden.MigrationShootWorkers]; ok {
+		if err := json.Unmarshal([]byte(data), &workerMigrationInfo); err != nil {
+			return err
+		}
+	}
+
+	switch {
+	case in.Spec.Cloud.AWS != nil:
+		out.Spec.Provider.Type = "aws"
+
+		extensionsScheme := runtime.NewScheme()
+		if err := awsinstall.AddToScheme(extensionsScheme); err != nil {
+			return err
+		}
+		decoder := serializer.NewCodecFactory(extensionsScheme).UniversalDecoder()
+
+		infrastructureConfig := &awsv1alpha1.InfrastructureConfig{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: awsv1alpha1.SchemeGroupVersion.String(),
+				Kind:       "InfrastructureConfig",
+			},
+		}
+		var infrastructureConfigOrigin string
+		if out.Spec.Provider.InfrastructureConfig != nil {
```

Comment:
As discussed, this can never be true when converting from `v1beta1` -> `internal`.

Reply:
Very nice feedback, thank you very much. I removed it entirely, unit tests + manual tests are still passing. Great! PTAL again :)